5calls/app/src/main/java/org/a5calls/android/a5calls/util/ScriptReplacements.java

```diff
@@ -0,0 +1,75 @@
+package org.a5calls.android.a5calls.util;
+
+import android.content.Context;
+import android.text.TextUtils;
+
+import androidx.annotation.Nullable;
+
+import org.a5calls.android.a5calls.R;
+import org.a5calls.android.a5calls.model.Contact;
+
+import java.util.regex.Pattern;
+
+public class ScriptReplacements {
+    private static final Pattern HOUSE_INTRO_PATTERN = Pattern.compile("\\*{2}WHEN CALLING HOUSE:\\*{2}\\n");
+    private static final Pattern SENATE_INTRO_PATTERN = Pattern.compile("\\*{2}WHEN CALLING SENATE:\\*{2}\\n");
+    private static final Pattern CONTACT_NAME_PATTERN = Pattern.compile("\\[REP/SEN NAME]|\\[SENATOR/REP NAME]|\\[SENATOR NAME]|\\[REPRESENTATIVE NAME]");
+    private static final Pattern LOCATION_PATTERN = Pattern.compile("\\[CITY,\\s?ZIP]|\\[CITY,\\s?STATE]");
+
+
+    public static String replacing(Context context, String script, Contact contact, @Nullable String location) {
+        String replacedScript = chooseSubscript(script, contact);
+        replacedScript = replacingContact(context, script, contact);
+
+        if (!TextUtils.isEmpty(location)) {
+            replacedScript = replacingLocation(replacedScript, location);
+        }
+
+        return replacedScript;
+    }
+
+    private static String chooseSubscript(String script, Contact contact) {
+        String area = contact.area;
+
+        if (area.equals("US House") || area.equals("House")) {
```

Comment:
nit: please make these strings into `private static final Strings` at the top of the file, it'll make them easier to find / modify if needed.